src/app/extensions/copilot/shared/copilot/copilot.component.ts

```diff
@@ -0,0 +1,253 @@
+import { Router } from '@angular/router';
+import { TranslateService } from '@ngx-translate/core';
+import { merge } from 'lodash-es';
+import { combineLatest } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+
+import { AppFacade } from 'ish-core/facades/app.facade';
+import { ShoppingFacade } from 'ish-core/facades/shopping.facade';
+import { GenerateLazyComponent } from 'ish-core/utils/module-loader/generate-lazy-component.decorator';
+import { ScriptLoaderService } from 'ish-core/utils/script-loader/script-loader.service';
+
+import { CompareFacade } from '../../../compare/facades/compare.facade';
+import { CopilotFacade } from '../../facades/copilot.facade';
+import { ChatbotMessage, ChatbotToolCall } from '../../models/copilot-chatbot/copilot-chatbot.model';
+import { CopilotConfig } from '../../models/copilot-config/copilot-config.model';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- allows access to Flowise Chatbot js functionality */
+declare let Chatbot: any;
+
+/**
+ * The Copilot Component
+ *
+ * This component initializes the Flowise Chatbot and provides the necessary configuration and localization data.
+ * It also handles the communication between the Chatbot and the PWA.
+ *
+ * @example
+ * <ish-lazy-copilot />
+ */
+@Component({
+  selector: 'ish-copilot',
+  template: '',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+@GenerateLazyComponent()
+export class CopilotComponent {
+  private destroyRef = inject(DestroyRef);
+
+  constructor(
+    private copilotFacade: CopilotFacade,
+    private scriptLoader: ScriptLoaderService,
+    private translateService: TranslateService,
+    private ngZone: NgZone,
+    private router: Router,
+    private appFacade: AppFacade,
+    private shoppingFacade: ShoppingFacade,
+    private compareFacade: CompareFacade
+  ) {
+    // afterNextRender = only rendered in browser
+    afterNextRender(() => {
+      this.loadCopilot();
+    });
+  }
+
+  /**
+   * Load the Flowise Chatbot functionality and initialize it with the fetched configuration and localization data.
+   */
+  private loadCopilot() {
+    combineLatest([
+      this.copilotFacade.copilotConfiguration$.pipe(
+        switchMap(config => this.scriptLoader.load(config.copilotUIFile, 'module').pipe(map(() => config)))
+      ),
+      this.translateService.get([
+        'copilot.disclaimer.buttonText',
+        'copilot.disclaimer.message',
+        'copilot.disclaimer.title',
+        'copilot.footer.company_link',
+        'copilot.footer.company',
+        'copilot.footer.text',
+        'copilot.max_chars_warning_message',
+        'copilot.placeholder',
+        'copilot.starter_prompt_1',
+        'copilot.starter_prompt_2',
+        'copilot.title',
+        'copilot.welcome_message',
+      ]),
+      this.appFacade.getRestEndpointWithContext$,
+      this.appFacade.currentLocale$,
+    ])
+      .pipe(takeUntilDestroyed(this.destroyRef))
```

Comment:
add take(1) here? Or can there be a situation the initializeCopilot has to be triggered more than once?

Reply:
You are right, this should only be loaded once.